CONTRIBUTING.md

```diff
@@ -0,0 +1,55 @@
+# Contributing
+
+## Adding a new rule
+New rules can be implemented in two ways:
+- as a `gosec.Rule` -- these define an arbitrary function which will be called on every AST node in the analyzed file, and are appropriate for rules that mostly need to reason about a single statement.
+- as an Analyzer -- these can operate on the entire program, and receive an [SSA](https://pkg.go.dev/golang.org/x/tools/go/ssa) representation of the program. This type of rule is useful when you need to perform a more complex analysis that requires a great deal of program context. 
+
+### Adding a gosec.Rule
+1. Copy an existing rule file as a starting point-- `./rules/unsafe.go` is a good option, as it implements a very simple rule with no additional supporting logic. Put the copied file in the `./rules/` directory.
+2. Change the name of the rule constructor function and of the types in the rule file you've copied so they will be unique.
+3. Edit the `Generate` function in `./rules/rulelist.go` to include your rule.
+4. Use `make` to compile `gosec`. The binary will now contain your rule.
```

Comment:
There is an extra step here. A mapping is needed from RuleID to CWE ID in https://github.com/securego/gosec/blob/329cad89ee05f29dc8ce797823a10960e558cf03/issue/issue.go#L56.

In case there isn't any appropriate CWE already, this needs to be added in https://github.com/securego/gosec/blob/329cad89ee05f29dc8ce797823a10960e558cf03/cwe/data.go#L20.

Reply:
Added!